src/newick.py

```diff
@@ -421,31 +421,128 @@ def write(tree, fname, encoding='utf8'):
         dump(tree, fp)
 
 
-def _parse_name_and_length(s):
-    l = None
-    if ':' in s:
-        s, l = s.split(':', 1)
-    return s or None, l or None
+def _parse_name(s, offset, node):
+    count = _count_spaces(s, offset)
+    match = COMMENT.search(s, offset + count)
+    if match:
+        name = s[offset + count:match.start()]
+        node.comment = comment = match.group(0)
```

Comment:
So turning `comment` into a property with appriate getter would already make this flexible enough, right? Derived node classes could then just overwrite a `parse_comment` or `parse_annotation` method an be done.